fsharp/FSharpParser.g4

```diff
@@ -0,0 +1,303 @@
+parser grammar FSharpParser;
+
+options {
+    tokenVocab=FSharpLexer;
+}
+
+// rules
+dot:DOT;
+
+identifier: IDENTIFIER;
+
+dotIentifier: identifier (dot identifier)*;
+
+int: INT;
+
+float: FLOAT;
+
+unit: UNIT;
+
+bool: BOOL;
+
+char: CHAR;
+
+underscore: MISSING_ARG MISSING_ARG?;
+
+interpolationSign: INTERPOLATIONSIGN; // don't add to expr
+
+dollar: DOLLAR; //don't add
+
+string: STRING;
+
+interpolated_string: INTERPOLATED_STRING;
+
+attribute: OPEN_BRACKET LESS dotIentifier GREATER CLOSE_BRACKET;
+
+generic: LESS dotIentifier GREATER;
+
+round_brackets: OPEN_PAREN (expression+ COMMA?)+ CLOSE_PAREN;
+
+rec: REC;
+
+public: PUBLIC;
+
+private: PRIVATE;
+
+internal: INTERNAL;
+
+mutable: MUTABLE;
+
+// let: LET ;
+
+let_fun: LET (mutable|internal|rec|public|private)* identifier (mutable|internal|rec|public|private)* ((identifier typezation?)|unit|round_brackets)+
+    typezation? equal expression;
+
+let_var: LET (mutable|internal|rec|public|private)* identifier typezation? (equal expression)?;
```

Comment:
I suggest extracting `mutable | internal | rec | public | private` to a separated rule:

```suggestion
let_fun: LET modifier* identifier modifier* (identifier typezation? | unit | round_brackets)+ typezation? equal expression;

let_var: LET modifier* identifier typezation? (equal expression)?;

modifier: mutable | internal | rec | public | private;
```